skills/studio/scripts/studio/cli.py

```
@@ -339,8 +342,37 @@ def _main_impl(argv_list: List[str]) -> int:
+        # decision id to this run so a later run/event can't inherit a stale id.
+        try:
+            decision_log.record_invocation(
+                cmd,
+                exit_code=handler_result,
+                duration_ms=int((perf_counter() - started) * 1000),
+                args_shape={"argc": len(rest),
+                            "flags": sum(1 for arg in rest if arg.startswith("-"))},
+                decision_id=decision_id,
+            )
+            decision_log.set_current_decision_id("")
+        except Exception as exc:  # pylint: disable=broad-except
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)